### PR TITLE
Adding command to convert environment file.

### PR DESCRIPTION
**Pull request by maxime-rainville**
I've added a command to convert an environment file from SS3 to SS4.
```bash
php bin/upgrade-code environment -d /var/www/
```

The command attempts to do basic checks on the environment file. But it's not meant to be 100% bullet proof. If your environment file contains mildly weird things, you'll get a warning. But it will still be converted.

On a side note, I've also refactored a bit of logic that was used by all the commands into a trait.

# Related issue
* [#59 Rewrite _ss_environment to .env in 4.x upgrades](https://github.com/silverstripe/silverstripe-upgrader/issues/59)

### CODE REVIEW

**Comment by maxime-rainville** [comment]
* I've created a `DotEnvLoader` to handle reading any current .env file and writing changes.
* `EnvParser` has been renamed to `LegacyEnvParser` to avoid confusion with `DotEnvLoader`.
* I've split the README into smaller file for readability.

File: src/Util/DotEnvLoader.php [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+    /**
+     * Full path to an env file that is read (if available). That's also where the results will be written.
+     * @var string
+     */
+    private $envFilePath;
+
+    /**
+     * Existing content of the .env file.
+     * @var [type]
```
Comment (tractorcow) [inline src/Util/DotEnvLoader.php]: Missing actual type here. ;)

File: src/Util/DotEnvLoader.php [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+     * @param string $envFilePath File Path to `.env` file
+     * @param array  $consts List of variables that should be added to envrionment file.
+     */
+    public function __construct(string $envFilePath, array $consts)
+    {
+        $this->envFilePath = $envFilePath;
+
+        $this->inputContent = $this->readFile();
+        $outputConsts = $this->parseFile($consts);
+        $this->outputContent = $this->buildOutput($outputConsts);
```
Comment (tractorcow) [inline src/Util/DotEnvLoader.php]: Should we merge both input and output env files? If we have a var in `.env` (and maybe not in `_ss_environment.php`) maybe it should still be in the output and not removed? It's probably ok if `_ss_environment.php` has higher priority (or whichever you want to do).

File: src/Util/DotEnvLoader.php [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+    /**
+     * @param string $envFilePath File Path to `.env` file
+     * @param array  $consts List of variables that should be added to envrionment file.
+     */
+    public function __construct(string $envFilePath, array $consts)
+    {
+        $this->envFilePath = $envFilePath;
+
+        $this->inputContent = $this->readFile();
```
Comment (tractorcow) [inline src/Util/DotEnvLoader.php]: Given that the constructor references almost exclusively private methods, think about how someone extending this class could want to modify this. Not so much a problem in this case, but more in the future when we are developing new core classes. :)

**Pull request merged**